net/netutil/netutil.go

```diff
@@ -116,3 +126,195 @@ func (w wrappedConn) Write(bs []byte) (int, error) {
 func (w wrappedConn) Close() error {
 	return w.rwc.Close()
 }
+
+// protocolsRequiredForForwarding reports whether IPv4 and/or IPv6 protocols are
+// required to forward the specified routes.
+// The state param must be specified.
+func protocolsRequiredForForwarding(routes []netaddr.IPPrefix, state *interfaces.State) (v4, v6 bool) {
+	if len(routes) == 0 {
+		// Nothing to route, so no need to warn.
+		return false, false
+	}
+
+	localIPs := make(map[netaddr.IP]bool)
+	for _, addrs := range state.InterfaceIPs {
+		for _, pfx := range addrs {
+			localIPs[pfx.IP()] = true
+		}
+	}
+
+	for _, r := range routes {
+		// It's possible to advertise a route to one of the local
+		// machine's local IPs. IP forwarding isn't required for this
+		// to work, so we shouldn't warn for such exports.
+		if r.IsSingleIP() && localIPs[r.IP()] {
```

Comment:
what about IsSingleIP && IsTailscaleIP? Does `routes` include `AllowedIPs` which includes the Tailscale self IPs? Document?

Reply:
Documented that this should only be `--advertised-routes`.